lib/rules/object-curly-spacing.js

```diff
@@ -121,9 +121,10 @@ module.exports = function(context) {
      * @returns {void}
      */
     function validateBraceSpacing(node, first, second, penultimate, last) {
+        var penultimateType = sourceCode.getNodeByRangeIndex(penultimate.start).type;
```

Comment:
This may be expensive so deferring the actual node getting to the very last moment in the boolean expressions below might be a good idea.